### DISCUSSION TITLE
Encontrar casos de uso

### DISCUSSION THREAD

**Discussion by alvarolostal** [DavidGarciaCosta/24-25-IdSw1-SDR#6] [General]
Hola, @mmasias.

Estamos teniendo problemas a la hora de [encontrar los casos de uso](https://github.com/DavidGarciaCosta/24-25-IdSw1-SDR/blob/main/documentos/CdU/ActoresCasosDeUso.md) de nuestro proyecto. Hemos hecho varias aproximaciones que no han terminado de convencernos; y al no tener bien el primer paso, encadenamos errores en todo el proceso.

¿Podrías mandarnos una aproximación de la idea que tienes sobre el paso de encontrar casos de uso y actores?

Muchas gracias.

**Comment by mmasias** [comment]
> Hola, @mmasias.
> 
> Estamos teniendo problemas a la hora de [encontrar los casos de uso](https://github.com/DavidGarciaCosta/24-25-IdSw1-SDR/blob/main/documentos/CdU/ActoresCasosDeUso.md) de nuestro proyecto. Hemos hecho varias aproximaciones que no han terminado de convencernos; y al no tener bien el primer paso, encadenamos errores en todo el proceso.
> 
> ¿Podrías mandarnos una aproximación de la idea que tienes sobre el paso de encontrar casos de uso y actores?
> 
> Muchas gracias.

- Un paso: plantearse qué nombre tienen, en el dominio, cada una de las operaciones CRUD de las entidades principales. 
- Las metodologías ágiles suelen plantear un "completa la oración": "como [ACTOR] quiero [ACCION] para [ALGUN_BENEFICIO]"

Si tu dominio tiene, por ejemplo, una ficha de datos, el llenar esa ficha de datos sería un caso de uso. Y a ese caso de uso podrias llegar con cualquiera de las dos aproximaciones anteriores.
